### ISSUE TITLE
Uneven class distribution, object detection

### ISSUE THREAD

**Issue by RoytenBerge** [tensorflow/models#4732]
What is the top-level directory of the model you are using
tensorflow/models/research
Have I written custom code
No
OS Platform and Distribution
Windows 10
TensorFlow installed from
Pip install
TensorFlow version
up-to-date 
Bazel version
CUDA/cuDNN version
not in use
GPU model and memory
not used
Exact command to reproduce
python train.py --logtostderr --train_dir=trainingBigSet/ --pipeline_config_path=object_detection/training/pipeline.config

------------------------------------------------------------------------------------------------------------

I am running the faster_rcnn_inception_v2_coco_2018_01_28 neural network on my home made dataset. It is not possible (due to my lack of data) to make an evenly divided dataset. Therefor I was wondering if there is a way of adding more weight to the classes that occur less in my dataset? 

I was thinking about multiplying the loss per class with a different weight, but i don't exactly know how to implement this. 

Your help is very much appreciated!

Kind regards,

Roy

--update
After further research I assume that I should edit some value in the faster_rcnn_meta_arch.py script:
```
def loss(self, prediction_dict, true_image_shapes, scope=None):
    """Compute scalar loss tensors given prediction tensors.

    If number_of_stages=1, only RPN related losses are computed (i.e.,
    `rpn_localization_loss` and `rpn_objectness_loss`).  Otherwise all
    losses are computed.

    Args:
      prediction_dict: a dictionary holding prediction tensors (see the
        documentation for the predict method.  If number_of_stages=1, we
        expect prediction_dict to contain `rpn_box_encodings`,
        `rpn_objectness_predictions_with_background`, `rpn_features_to_crop`,
        `image_shape`, and `anchors` fields.  Otherwise we expect
        prediction_dict to additionally contain `refined_box_encodings`,
        `class_predictions_with_background`, `num_proposals`, and
        `proposal_boxes` fields.
      true_image_shapes: int32 tensor of shape [batch, 3] where each row is
        of the form [height, width, channels] indicating the shapes
        of true images in the resized images, as resized images can be padded
        with zeros.
      scope: Optional scope name.

    Returns:
      a dictionary mapping loss keys (`first_stage_localization_loss`,
        `first_stage_objectness_loss`, 'second_stage_localization_loss',
        'second_stage_classification_loss') to scalar tensors representing
        corresponding loss values.
    """
    with tf.name_scope(scope, 'Loss', prediction_dict.values()):
      (groundtruth_boxlists, groundtruth_classes_with_background_list,
       groundtruth_masks_list) = self._format_groundtruth_data(
           true_image_shapes)
      loss_dict = self._loss_rpn(
          prediction_dict['rpn_box_encodings'],
          prediction_dict['rpn_objectness_predictions_with_background'],
          prediction_dict['anchors'],
          groundtruth_boxlists,
          groundtruth_classes_with_background_list)
      if self._number_of_stages > 1:
        loss_dict.update(
            self._loss_box_classifier(
                prediction_dict['refined_box_encodings'],
                prediction_dict['class_predictions_with_background'],
                prediction_dict['proposal_boxes'],
                prediction_dict['num_proposals'],
                groundtruth_boxlists,
                groundtruth_classes_with_background_list,
                prediction_dict['image_shape'],
                prediction_dict.get('mask_predictions'),
                groundtruth_masks_list,
            ))
    return loss_dict

```
There is a function def _loss_box_classifier that holds the classification loss calculation:
`      classification_loss = tf.multiply(self._second_stage_cls_loss_weight,
                                        second_stage_cls_loss,
                                        name='classification_loss')`
On line 1828. This might be the line that needs to be multiplied with the class multiplier?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by kulkarnivishal** [comment]
Any update/solution to this?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by kulkarnivishal** [comment]
Yes, I still need help on this
